### PR TITLE
Remove unnecessary requires

### PR DESCRIPTION
**Pull request by alysbrooks**
Added some comments about some edge cases.

Unnecessary `require`s aren't a major issue, but there are advantages to cleaning them up. This PR silences some warnings from `clj-kondo`, makes the namespaces easier to understand, and may avoid some circular imports in the future.

### CODE REVIEW

File: src/kaocha/runner.clj [hunk 1]
```diff
@@ -15,7 +15,7 @@
             [kaocha.output :as output]
             [kaocha.plugin :as plugin]
             [kaocha.result :as result]
-            [kaocha.specs :as specs]
+            #_[kaocha.specs :as specs] ;; required for some reason?
```
Comment (borkdude) [inline src/kaocha/runner.clj:18]: If you require namespaces for side effect, you can leave out the alias and clj-kondo won't report it as unused.

Comment (plexus) [reply src/kaocha/runner.clj]: When namespaces are required for side-effects, the convention we use is to `(require '...)` them below the ns form.

Comment (borkdude) [reply src/kaocha/runner.clj]: This also works (for clj-kondo)

**Comment by alysbrooks** [comment]
This seems slightly more prone to breaking things not covered by tests, so I figure we should wait until #413 is merged.

**Comment by alysbrooks** [comment]
Released in [v1.84.1335](https://github.com/lambdaisland/kaocha/releases/tag/v1.84.1335)

```clj
[lambdaisland/kaocha "1.84.1335"]                 ;; deps.edn
{lambdaisland/kaocha {:mvn/version "1.84.1335"}}  ;; project.clj
```

**Pull request merged**